Enable fractional coordinates for Pointer Events and Touch Events

lilyspiniolas opened the pull request:
#### 8dc5033bb0ead7dc8b4cabba88a45f5b1cba9c88
<pre>
Enable fractional coordinates for Pointer Events and Touch Events
<a href="https://bugs.webkit.org/show_bug.cgi?id=133180">https://bugs.webkit.org/show_bug.cgi?id=133180</a>
<a href="https://rdar.apple.com/9564176">rdar://9564176</a>

Reviewed by Abrar Rahman Protyasha.

Pointer Events and Touch Events now expose fractional coordinate values for
clientX/clientY, pageX/pageY, offsetX/offsetY, and screenX/screenY.
Mouse Events still report whole values. These changes affect all platforms.

For pointer events, events with type `click`, `auxclick`, or `contextmenu`
report floored values.

Create new versions of `DoublePoint`, `DoubleRect`, and `DoubleSize` in WebCore.
These versions retain all existing functionality, and include new additions which
allow the types to be used for pointer event location calculations.

Test coverage:
* Fractional coordinates for mouse and pen pointer events from within a subframe are tested by
  `imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates.html`
  which now passes on macOS, GTK, WPE, and Windows. This test also ensures that
  click events are not fractional.

* Fractional coordinates for touch events from within a subframe are tested by
  `fast/events/touch/touch-fractional-coordinates.html`.

* Fractional coordinates for mouse pointer events on iOS from within a subframe are tested by
  new API tests `FractionalCoordinatesInSimpleIFrame`, `FractionalCoordinatesInRotatedIFrame`
  &amp; `FractionalCoordinatesInScaledIFrame`.

* `imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates.html` now
  checks both client and page coordinates to increase the robustness of the test.

These changes build upon the following proposed patch by Sun Shin (@xingri):
<a href="https://github.com/WebKit/WebKit/pull/31867">https://github.com/WebKit/WebKit/pull/31867</a>

(Additionally, updated `setContentsToPlatformLayer` to take a `PlatformLayerContainer` instead of a
`PlatformLayer *` due to a safer-cpp issue in order to improve safety.)

* LayoutTests/fast/events/constructors/mouse-event-constructor-expected.txt:
* LayoutTests/fast/events/constructors/mouse-event-constructor.html:
* LayoutTests/fast/events/constructors/wheel-event-constructor-expected.txt:
* LayoutTests/fast/events/constructors/wheel-event-constructor.html:
* LayoutTests/fast/events/touch/ios/touches-client-coords-after-zoom.html:
* LayoutTests/fast/events/touch/resources/frame-touchevent-forwarder.html: Added.
* LayoutTests/fast/events/touch/touch-coords-in-zoom-and-scroll-expected.txt:
* LayoutTests/fast/events/touch/touch-coords-in-zoom-and-scroll.html:
* LayoutTests/fast/events/touch/touch-fractional-coordinates-expected.txt: Added.
* LayoutTests/fast/events/touch/touch-fractional-coordinates.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates.html:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates_mouse-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates_pen-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates_untrusted-expected.txt:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/ios/fast/events/touch/ios/touches-client-coords-after-zoom-expected.txt:
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Document+Touch.idl:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::showPlaybackTargetPicker):
* Source/WebCore/dom/DocumentTouch.cpp:
(WebCore::DocumentTouch::createTouch):
* Source/WebCore/dom/DocumentTouch.h:
* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::create):
(WebCore::MouseEvent::MouseEvent):
(WebCore::MouseEvent::initMouseEvent):
* Source/WebCore/dom/MouseEvent.h:
* Source/WebCore/dom/MouseEvent.idl:
* Source/WebCore/dom/MouseEventInit.h:
* Source/WebCore/dom/MouseEventInit.idl:
* Source/WebCore/dom/MouseRelatedEvent.cpp:
(WebCore::MouseRelatedEvent::MouseRelatedEvent):
(WebCore::MouseRelatedEvent::init):
(WebCore::MouseRelatedEvent::initCoordinates):
(WebCore::MouseRelatedEvent::pagePointToClientPoint):
(WebCore::MouseRelatedEvent::pagePointToAbsolutePoint):
(WebCore::MouseRelatedEvent::computeRelativePosition):
(WebCore::MouseRelatedEvent::locationInRootViewCoordinates const):
(WebCore::MouseRelatedEvent::offsetX):
(WebCore::MouseRelatedEvent::offsetY):
(WebCore::MouseRelatedEvent::pageX const):
(WebCore::MouseRelatedEvent::pageY const):
(WebCore::finiteValue):
(WebCore::MouseRelatedEvent::screenX const):
(WebCore::MouseRelatedEvent::screenY const):
(WebCore::MouseRelatedEvent::clientX const):
(WebCore::MouseRelatedEvent::clientY const):
* Source/WebCore/dom/MouseRelatedEvent.h:
(WebCore::MouseRelatedEvent::screenLocation const):
(WebCore::MouseRelatedEvent::windowLocation const):
(WebCore::MouseRelatedEvent::clientLocation const):
(WebCore::MouseRelatedEvent::absoluteLocation const):
(WebCore::MouseRelatedEvent::hasCachedRelativePosition const):
(WebCore::MouseRelatedEvent::offsetLocation const):
(WebCore::MouseRelatedEvent::pageLocation const):
(WebCore::MouseRelatedEvent::setScreenLocation):
(WebCore::MouseRelatedEvent::screenX const): Deleted.
(WebCore::MouseRelatedEvent::screenY const): Deleted.
(WebCore::MouseRelatedEvent::clientX const): Deleted.
(WebCore::MouseRelatedEvent::clientY const): Deleted.
* Source/WebCore/dom/PointerEvent.cpp:
(WebCore::PointerEvent::PointerEvent):
(WebCore::m_fractionalCoordinatesAllowed):
(WebCore::PointerEvent::fractionalCoordinatesAllowedForType):
(WebCore::PointerEvent::adjustedCoordinateForType const):
(WebCore::PointerEvent::offsetX):
(WebCore::PointerEvent::offsetY):
(WebCore::m_isPrimary): Deleted.
* Source/WebCore/dom/PointerEvent.h:
(WebCore::PointerEvent::create):
(WebCore::PointerEvent::fractionalCoordinatesAllowed const):
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/dom/SimulatedClick.cpp:
* Source/WebCore/dom/Touch.cpp:
(WebCore::contentsOffset):
(WebCore::scaledLocation):
(WebCore::Touch::Touch):
(WebCore::Touch::cloneWithNewTarget const):
(WebCore::contentsX): Deleted.
(WebCore::contentsY): Deleted.
* Source/WebCore/dom/Touch.h:
(WebCore::Touch::create):
(WebCore::Touch::clientX const):
(WebCore::Touch::clientY const):
(WebCore::Touch::screenX const):
(WebCore::Touch::screenY const):
(WebCore::Touch::pageX const):
(WebCore::Touch::pageY const):
(WebCore::Touch::webkitRadiusX const):
(WebCore::Touch::webkitRadiusY const):
(WebCore::Touch::absoluteLocation const):
* Source/WebCore/dom/Touch.idl:
* Source/WebCore/dom/TouchEvent.cpp:
(WebCore::TouchEvent::TouchEvent):
* Source/WebCore/dom/TouchEvent.h:
* Source/WebCore/dom/UIEvent.cpp:
(WebCore::UIEvent::screenX const):
(WebCore::UIEvent::screenY const):
(WebCore::UIEvent::pageX const):
(WebCore::UIEvent::pageY const):
(WebCore::UIEvent::clientX const):
(WebCore::UIEvent::clientY const):
* Source/WebCore/dom/UIEvent.h:
* Source/WebCore/dom/UIEvent.idl:
* Source/WebCore/dom/ios/PointerEventIOS.cpp:
(WebCore::PointerEvent::create):
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/dom/wpe/PointerEventWPE.cpp:
(WebCore::PointerEvent::create):
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/html/CheckboxInputType.cpp:
(WebCore::CheckboxInputType::handleMouseDownEvent):
(WebCore::CheckboxInputType::handleMouseMoveEvent):
(WebCore::CheckboxInputType::handleTouchEvent):
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleMouseDownEvent):
(WebCore::RangeInputType::handleTouchEvent):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::defaultEventHandler):
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::InspectorFrontendHost::dispatchEventAsContextMenuEvent):
* Source/WebCore/page/AutoscrollController.cpp:
(WebCore::AutoscrollController::autoscrollTimerFired):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::maybeCreateContextMenu):
* Source/WebCore/page/DebugPageOverlays.cpp:
(WebCore::InteractionRegionOverlay::mouseEvent):
* Source/WebCore/page/DragController.cpp:
(WebCore::DragController::startDrag):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::SyntheticTouchPoint::SyntheticTouchPoint):
(WebCore::EventHandler::handleMousePressEventSingleClick):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::eventMayStartDrag const):
(WebCore::EventHandler::updateSelectionForMouseDrag):
(WebCore::EventHandler::startPanScrolling):
(WebCore::EventHandler::lastKnownMousePosition const):
(WebCore::EventHandler::updateCursor):
(WebCore::documentPointForWindowPoint):
(WebCore::scrollbarForMouseEvent):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseReleaseEvent):
(WebCore::EventHandler::dispatchDragEvent):
(WebCore::EventHandler::updateDragAndDrop):
(WebCore::EventHandler::prepareMouseEvent):
(WebCore::EventHandler::dispatchMouseEvent):
(WebCore::EventHandler::targetPositionInWindowForSelectionAutoscroll const):
(WebCore::EventHandler::handleWheelEventInternal):
(WebCore::EventHandler::sendContextMenuEvent):
(WebCore::EventHandler::dispatchFakeMouseMoveEventSoonInQuad):
(WebCore::EventHandler::hoverTimerFired):
(WebCore::EventHandler::handleDrag):
(WebCore::EventHandler::handleTouchEvent):
(WebCore::EventHandler::setLastKnownMousePosition):
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::convertFromContainingViewToRenderer const):
(WebCore::FrameView::convertFromContainingView const):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::lastKnownMousePositionInView const):
(WebCore::LocalFrameView::absoluteToDocumentPoint const):
(WebCore::LocalFrameView::documentToClientPoint const):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/PageOverlay.cpp:
(WebCore::PageOverlay::mouseEvent):
* Source/WebCore/page/PointerCaptureController.cpp:
(WebCore::PointerCaptureController::dispatchOverOrOutEvent):
(WebCore::PointerCaptureController::dispatchEnterOrLeaveEvent):
(WebCore::PointerCaptureController::dispatchEventForTouchAtIndex):
* Source/WebCore/page/PointerCaptureController.h:
* Source/WebCore/page/RemoteFrameGeometryTransformer.cpp:
(WebCore::RemoteFrameGeometryTransformer::transformToRemoteFrameCoordinates const):
* Source/WebCore/page/RemoteFrameGeometryTransformer.h:
* Source/WebCore/page/ResourceUsageOverlay.cpp:
(WebCore::ResourceUsageOverlay::mouseEvent):
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::tryToBeginDragAtPoint):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::targetPositionInWindowForSelectionAutoscroll const):
* Source/WebCore/page/mac/ImageOverlayControllerMac.mm:
(WebCore::ImageOverlayController::platformHandleMouseEvent):
* Source/WebCore/page/mac/ServicesOverlayController.mm:
(WebCore::ServicesOverlayController::mouseEvent):
* Source/WebCore/platform/PlatformMouseEvent.h:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
(WebCore::PlatformMouseEvent::position const):
(WebCore::PlatformMouseEvent::globalPosition const):
(WebCore::PlatformMouseEvent::movementDelta const):
(WebCore::PlatformMouseEvent::unadjustedMovementDelta const):
* Source/WebCore/platform/PlatformTouchPoint.h:
(WebCore::PlatformTouchPoint::PlatformTouchPoint):
(WebCore::PlatformTouchPoint::screenPos const):
(WebCore::PlatformTouchPoint::pos const):
(WebCore::PlatformTouchPoint::radius const):
(WebCore::PlatformTouchPoint::radiusX const): Deleted.
(WebCore::PlatformTouchPoint::radiusY const): Deleted.
* Source/WebCore/platform/ScrollView.cpp:
(WebCore::ScrollView::viewToContents const):
(WebCore::ScrollView::contentsToView const):
(WebCore::ScrollView::rootViewToContents const):
(WebCore::ScrollView::contentsToRootView const):
(WebCore::ScrollView::windowToContents const):
(WebCore::ScrollView::convertChildToSelf const):
(WebCore::ScrollView::convertSelfToChild const):
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Scrollbar.cpp:
(WebCore::Scrollbar::mouseMoved):
(WebCore::Scrollbar::mouseUp):
(WebCore::Scrollbar::mouseDown):
* Source/WebCore/platform/Widget.cpp:
(WebCore::Widget::convertToRootView const):
(WebCore::Widget::convertFromRootView const):
(WebCore::Widget::convertFromContainingWindow const):
(WebCore::Widget::convertToContainingView const):
(WebCore::Widget::convertFromContainingView const):
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/graphics/DoublePoint.cpp: Copied from Source/WebCore/dom/DocumentTouch.h.
(WebCore::DoublePoint::DoublePoint):
(WebCore::DoublePoint::operator CGPoint const):
(WebCore::DoublePoint::toJSONObject const):
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/graphics/DoublePoint.h: Added.
(WebCore::DoublePoint::DoublePoint):
(WebCore::DoublePoint::zero):
(WebCore::DoublePoint::isZero const):
(WebCore::DoublePoint::x const):
(WebCore::DoublePoint::y const):
(WebCore::DoublePoint::move):
(WebCore::DoublePoint::moveBy):
(WebCore::DoublePoint::scale):
(WebCore::DoublePoint::scaled const):
(WebCore::DoublePoint::toCG const):
(WebCore::operator+):
(WebCore::operator-):
(WebCore::flooredIntPoint):
(WebCore::roundedIntPoint):
(WebCore::toDoubleSize):
(WTF::LogArgument&lt;WebCore::DoublePoint&gt;::toString):
* Source/WebCore/platform/graphics/DoubleRect.cpp: Copied from Source/WebCore/dom/MouseEventInit.h.
(WebCore::DoubleRect::DoubleRect):
(WebCore::DoubleRect::operator CGRect const):
* Source/WebCore/platform/graphics/DoubleRect.h: Copied from Source/WebCore/dom/DocumentTouch.h.
(WebCore::DoubleRect::DoubleRect):
(WebCore::DoubleRect::location const):
(WebCore::DoubleRect::size const):
(WebCore::DoubleRect::toCG const):
* Source/WebCore/platform/graphics/DoubleSize.cpp: Copied from Source/WebCore/dom/MouseEventInit.h.
(WebCore::DoubleSize::DoubleSize):
(WebCore::DoubleSize::operator CGSize const):
* Source/WebCore/platform/graphics/DoubleSize.h: Copied from Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.h.
(WebCore::DoubleSize::DoubleSize):
(WebCore::DoubleSize::width const):
(WebCore::DoubleSize::height const):
(WebCore::DoubleSize::toCG const):
(WebCore::DoubleSize::scaledBy const):
* Source/WebCore/platform/graphics/FloatPoint.h:
(WebCore::FloatPoint::FloatPoint):
(WebCore::FloatPoint::scaledBy const):
(WebCore::FloatPoint::operator DoublePoint const):
(WebCore::roundedIntPoint):
* Source/WebCore/platform/graphics/FloatSize.h:
(WebCore::FloatSize::scaledBy const):
(WebCore::FloatSize::operator DoubleSize const):
* Source/WebCore/platform/graphics/Font.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/graphics/LayoutPoint.h:
(WebCore::LayoutPoint::LayoutPoint):
(WebCore::LayoutPoint::operator DoublePoint const):
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::GraphicsLayerCA::setContentsToVideoElement):
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
* Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
(WebCore::Font::applyTransforms const):
* Source/WebCore/platform/ios/PlatformEventFactoryIOS.mm:
(WebCore::pointForEvent):
(WebCore::globalPointForEvent):
(WebCore::PlatformWheelEventBuilder::PlatformWheelEventBuilder):
* Source/WebCore/platform/ios/ScrollAnimatorIOS.mm:
(WebCore::ScrollAnimatorIOS::handleTouchEvent):
* Source/WebCore/platform/ios/WidgetIOS.mm:
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.h:
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::pointForEvent):
(WebCore::unadjustedMovementForEvent):
(WebCore::PlatformMouseEventBuilder::PlatformMouseEventBuilder):
(WebCore::PlatformWheelEventBuilder::PlatformWheelEventBuilder):
* Source/WebCore/platform/mac/WidgetMac.mm:
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::resize):
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::panScrollFromPoint):
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::panScroll):
(WebCore::RenderListBox::autoscroll):
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::absoluteToLocal const):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::absoluteToLocal):
* Source/WebCore/testing/MockPageOverlayClient.cpp:
(WebCore::MockPageOverlayClient::mouseEvent):
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.h:
(WebKit::CoreIPCNSShadow::CoreIPCNSShadow):
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.mm:
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.mm:
(WebKit::CoreIPCNSValue::toID const):
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionTab.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionTabParameters.h:
* Source/WebKit/Shared/Extensions/WebExtensionWindow.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionWindowParameters.h:
* Source/WebKit/Shared/NativeWebMouseEvent.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
(WebKit::WebKit2PlatformTouchPoint::WebKit2PlatformTouchPoint):
* Source/WebKit/Shared/WebMouseEvent.cpp:
(WebKit::WebMouseEvent::WebMouseEvent):
* Source/WebKit/Shared/WebMouseEvent.h:
(WebKit::WebMouseEvent::WebMouseEvent):
(WebKit::WebMouseEvent::position const):
(WebKit::WebMouseEvent::setPosition):
(WebKit::WebMouseEvent::globalPosition const):
(WebKit::WebMouseEvent::unadjustedMovementDelta const):
* Source/WebKit/Shared/WebPlatformTouchPoint.cpp:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
* Source/WebKit/Shared/WebTouchEvent.h:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
(WebKit::WebPlatformTouchPoint::locationInRootView const):
(WebKit::WebPlatformTouchPoint::locationInViewport const):
(WebKit::WebTouchEvent::WebTouchEvent):
(WebKit::WebTouchEvent::position const):
(WebKit::WebPlatformTouchPoint::screenPosition const):
(WebKit::WebPlatformTouchPoint::position const):
(WebKit::WebPlatformTouchPoint::radius const):
* Source/WebKit/Shared/graphics/DoubleGeometry.h: Removed.
* Source/WebKit/Shared/graphics/DoubleGeometry.serialization.in: Removed.
* Source/WebKit/Shared/gtk/NativeWebMouseEventGtk.cpp:
(WebKit::NativeWebMouseEvent::NativeWebMouseEvent):
* Source/WebKit/Shared/gtk/WebEventFactory.cpp:
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/Shared/gtk/WebEventFactory.h:
* Source/WebKit/Shared/ios/NativeWebMouseEventIOS.mm:
(WebKit::NativeWebMouseEvent::NativeWebMouseEvent):
* Source/WebKit/Shared/ios/NativeWebTouchEventIOS.mm:
(WebKit::positionForCGPoint):
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(webkitWebViewBaseButtonPressed):
(webkitWebViewBaseButtonReleased):
(webkitWebViewBaseEnter):
(webkitWebViewBaseMotion):
(webkitWebViewBaseLeave):
(webkitWebViewBaseSynthesizeMouseEvent):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateTouchEventTracking):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _handleTouchActionsForTouchEvent:]):
* Source/WebKit/UIProcess/ios/WKMouseInteraction.mm:
(-[WKMouseInteraction createMouseEventWithType:wasCancelled:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPITabsCocoa.mm:
(WebKit::toWebAPI):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePageOverlay.cpp:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::handleMouseEvent):
(WebKit::PDFPlugin::handleContextMenuEvent):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::lastKnownMousePositionInView const):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFDataDetectorOverlayController.mm:
(WebKit::PDFDataDetectorOverlayController::handleMouseEvent):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::handleMouseEvent):
(WebKit::UnifiedPDFPlugin::handleContextMenuEvent):
(WebKit::UnifiedPDFPlugin::createContextMenu const):
(WebKit::UnifiedPDFPlugin::handleSyntheticClick):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::handleContextMenuEvent):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::rectForElementAtInteractionLocation const):
(WebKit::WebPage::focusedElementInformation):
* Source/WebKit/WebProcess/WebPage/mac/PageBannerMac.mm:
(WebKit::PageBanner::mouseEvent):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::shouldDelayWindowOrderingEvent):
(WebKit::WebPage::requestAcceptsFirstMouse):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(TEST_F(iOSMouseSupport, FractionalCoordinatesInSimpleIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInRotatedIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInScaledIFrame)):

Canonical link: <a href="https://commits.webkit.org/298383@main">https://commits.webkit.org/298383@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c584cd1507bd1d92ae3a1f0ebd2d6ac237588fd4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115350 "5 style errors") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35044 "Build is in progress. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25546 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121440 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65932 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e8c07c34-8765-4b9f-b16e-b1d6bd9d8a34) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117239 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35703 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43616 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87642 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42349 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5b0e6b6d-7463-4439-b80c-99f628e6a92a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118298 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28467 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103547 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68037 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27626 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21668 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65100 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97853 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21782 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124607 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42289 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31672 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96424 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42657 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99735 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96211 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24477 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41433 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19293 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38220 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42172 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47725 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41677 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45006 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43401 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/dom/MouseRelatedEvent.cpp:
```diff
@@ -63,19 +62,29 @@ MouseRelatedEvent::MouseRelatedEvent(enum EventInterfaceType eventInterface, con
+
+    m_screenLocation = DoublePoint(screenX, screenY);
+
+    init(false, DoublePoint(0, 0));
```
We could add `static constexpr DoublePoint zero() { return DoublePoint(); }` in `DoublePoint.h` and use it here.

(`IntPoint::zero` is already a thing and should have been used here earlier)

pxlcoder commented on Source/WebCore/dom/MouseRelatedEvent.h:
```diff
@@ -45,70 +45,77 @@ class MouseRelatedEvent : public UIEventWithKeyState {
+    DoublePoint m_clientLocation;
+
+    // Expose these so PointerEvent can access it.
+    // FIXME: These probably should use a getter method
```
Is there a reason this FIXME isn't being addressed now? The current code would be against our style guide.

lilyspiniolas replied:
No, I just need to do some general cleanup still (including this)

pxlcoder commented on Source/WebCore/dom/Touch.h:
```diff
@@ -41,49 +42,49 @@ class LocalFrame;
-    int m_radiusX;
-    int m_radiusY;
+    // Position relative to the viewport in CSS px.
+    DoublePoint m_clientPos;
```
`m_clientPosition`, etc. style guide says to avoid short forms.

pxlcoder commented on Source/WebCore/inspector/InspectorFrontendHost.cpp:
```diff
@@ -616,7 +616,7 @@ void InspectorFrontendHost::dispatchEventAsContextMenuEvent(Event& event)
     auto& mouseEvent = downcast<MouseEvent>(event);
     LocalFrame& frame = *downcast<Node>(mouseEvent.target())->document().frame();
-    LayoutPoint location = mouseEvent.absoluteLocation();
+    LayoutPoint location = LayoutPoint(mouseEvent.absoluteLocation());
```
Unclear why this change is necessary?

lilyspiniolas replied:
absoluteLocation() returns a DoublePoint now. The LayoutPoint constructor which takes a DoublePoint is explicit since it is lossy.

pxlcoder replied:
Let's do `auto location = ...` then.

pxlcoder commented on Source/WebCore/platform/ios/WidgetIOS.mm:
```diff
@@ -296,6 +296,24 @@ static void safeRemoveFromSuperview(NSView *view)
+    BEGIN_BLOCK_OBJC_EXCEPTIONS
+    WAKScrollView *view = checked_objc_cast<WAKScrollView>(rootWidget->platformWidget());
+    NSPoint convertedPoint;
+    if (WAKView *documentView = [view documentView])
```
Let's use `RetainPtr` since this is new code.

pxlcoder commented on Source/WebCore/platform/graphics/DoubleSize.h:
```diff
@@ -25,79 +25,36 @@
-    double x { 0 };
-    double y { 0 };
-};
+class DoubleSize {
```
This PR is quite large as-is, so I would recommend splitting up any changes involving the moving of types from WebKit -> WebCore into a separate, first PR.

rr-codes commented on Source/WebCore/platform/graphics/DoublePoint.h:
```diff
@@ -0,0 +1,169 @@
+
+class IntSize;
+
+class DoublePoint {
```
does this need to be a separate type? can the existing FloatPoint type be templated to accept either a float or a double data type?

lilyspiniolas replied:
It could be, but if we go that route it would probably make sense to rename `FloatPoint` to something else, which would make this patch a lot larger than it already is. Maybe in a separate patch?

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates.html:
```diff
@@ -45,8 +45,10 @@
+              if ((event.clientX != Math.floor(event.clientX) ||
+                    event.clientY != Math.floor(event.clientY)) &&
+                    (event.pageX != Math.floor(event.pageX) ||
+                    event.pageY != Math.floor(event.pageY)))
```
```suggestion
              if ((event.clientX != Math.floor(event.clientX) || event.clientY != Math.floor(event.clientY))
                    && (event.pageX != Math.floor(event.pageX) || event.pageY != Math.floor(event.pageY)))
```
I find this easier to read

aprotyas reviewed (commented):
`#import <QuartzCore/CALayer.h>` to deal with the forward decl warning?

aprotyas commented on Source/WebCore/dom/DocumentTouch.cpp:
```diff
@@ -37,19 +37,38 @@
     if (!frame)
         frame = document.frame();
 
+    // Match behavior from when these types were integers, and avoid surprises from someone explicitly
+    // passing Infinity/NaN.
```
> and avoid surprises from someone explicitly passing Infinity/NaN.

What was the behavior before this patch? When these were inf/nan?

lilyspiniolas replied:
Always zero when passed `Infinity` or `NaN` via the JS constructor

aprotyas commented on Source/WebCore/dom/MouseEvent.cpp:
```diff
@@ -139,21 +148,30 @@ MouseEvent::MouseEvent(enum EventInterfaceType eventInterface, const AtomString&
     m_relatedTarget = relatedTarget;
 
-    initCoordinates(IntPoint(clientX, clientY));
+    initCoordinates(DoublePoint(clientX, clientY));
```
```suggestion
    initCoordinates({ clientX, clientY });
```

aprotyas commented on Source/WebCore/dom/MouseRelatedEvent.cpp:
```diff
@@ -52,30 +51,40 @@ MouseRelatedEvent::MouseRelatedEvent(enum EventInterfaceType eventInterface)
+        , m_movementY(movementY)
+        , m_windowLocation(windowLocation)
+        , m_screenLocation(screenLocation)
+        , m_isSimulated(isSimulated == IsSimulated::Yes)
```
Is this for style guide conformance?

lilyspiniolas replied:
Yeah, the style checker expects 8 spaces

aprotyas commented on Source/WebCore/dom/MouseRelatedEvent.cpp:
```diff
@@ -86,13 +95,13 @@ static inline bool isMoveEventType(const AtomString& eventType)
-            m_pageLocation = flooredLayoutPoint(documentPoint);
+            DoublePoint absolutePoint = frameView->windowToContents(windowLocation);
+            DoublePoint documentPoint = frameView->absoluteToDocumentPoint(absolutePoint);
+            m_pageLocation = DoublePoint(documentPoint);
```
You already have a DoublePoint so no need to create one again
```suggestion
            DoublePoint documentPoint = frameView->absoluteToDocumentPoint(absolutePoint);
            m_pageLocation = WTFMove(documentPoint);
```

aprotyas commented on Source/WebCore/dom/MouseRelatedEvent.cpp:
```diff
@@ -129,34 +138,35 @@ LocalFrameView* MouseRelatedEvent::frameViewFromWindowProxy(WindowProxy* windowP
     if (RefPtr frameView = frameViewFromWindowProxy(view()))
         documentToClientOffset = frameView->documentToClientOffset();
 
+    DoubleSize offset(documentToClientOffset.width(), documentToClientOffset.height());
```
Do you think it's useful to have an (explicit) `DoubleSize(FloatSize&&)` constructor? That way you could write:

```cpp
auto documentToClientOffset = [&] -> DoubleSize {
    if (RefPtr frameView = frameViewFromWindowProxy(view()))
        return frameView->documentToClientOffset();
    return { };
};
```

aprotyas commented on Source/WebCore/dom/PointerEvent.cpp:
```diff
@@ -293,4 +293,39 @@ void PointerEvent::receivedTarget()
+    return !(type() == eventNames().auxclickEvent
+        || type() == eventNames().clickEvent
+        || type() == eventNames().contextmenuEvent);
+}
```
This can probably be cached at construction time

aprotyas replied:
(I'm imagining there may be lots of `adjustedCoordinateForType` calls?)

aprotyas commented on Source/WebCore/dom/Touch.h:
```diff
@@ -41,52 +42,52 @@ class LocalFrame;
-            int screenX, int screenY, int pageX, int pageY,
-            int radiusX, int radiusY, float rotationAngle, float force);
+        const DoublePoint& screenPosition, const DoublePoint& pagePosition,
+            const DoubleSize& radius, float rotationAngle, float force);
```
Some spacing discontinuity?

aprotyas commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -3077,12 +3077,13 @@ bool EventHandler::dispatchMouseEvent(const AtomString& eventType, Node* targetN
     Ref frame = m_frame.get();
 
-    updateMouseEventTargetNode(eventType, targetNode, platformMouseEvent, fireMouseOverOut);
+    PlatformMouseEvent mouseEvent = platformMouseEvent;
```
Why did we copy the event?

lilyspiniolas replied:
I think this is left over from an earlier change I made and later removed, good catch

aprotyas commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -3625,12 +3627,12 @@ bool EventHandler::sendContextMenuEvent(const PlatformMouseEvent& event)
     m_mousePressed = false;
     bool swallowEvent;
-    LayoutPoint viewportPos = view->windowToContents(event.position());
+    LayoutPoint viewportPos = view->windowToContents(flooredIntPoint(event.position()));
```
nit: You could re-use `DoublePoint position = event.position();` here and again in a few lines?

aprotyas commented on Source/WebCore/page/FrameView.cpp:
```diff
@@ -242,6 +242,17 @@ FloatPoint FrameView::convertFromContainingViewToRenderer(const RenderElement* r
 
+DoublePoint FrameView::convertFromContainingViewToRenderer(const RenderElement* renderer, DoublePoint viewPoint) const
+{
+    auto point = viewPoint;
```
Why the copy?

aprotyas commented on Source/WebCore/page/RemoteFrameGeometryTransformer.cpp:
```diff
@@ -52,4 +52,9 @@ FloatPoint RemoteFrameGeometryTransformer::transformToRemoteFrameCoordinates(Flo
 
+DoublePoint RemoteFrameGeometryTransformer::transformToRemoteFrameCoordinates(DoublePoint pointInContents) const
+{
+    return Ref { m_remoteView }->rootViewToContents(Ref { m_localView }->contentsToRootView(pointInContents));
```
I think you don't need to Ref these here?

aprotyas commented on Source/WebCore/platform/graphics/DoublePoint.h:
```diff
@@ -0,0 +1,169 @@
+
+WEBCORE_EXPORT WTF::TextStream& operator<<(WTF::TextStream&, const DoublePoint&);
+
+}
```
nit
```suggestion
} // namespace WebCore
```

aprotyas commented on Source/WebCore/platform/ios/WidgetIOS.mm:
```diff
@@ -296,6 +296,24 @@ static void safeRemoveFromSuperview(NSView *view)
+        convertedPoint = IntPoint([documentView convertPoint:point fromView:nil]);
+    else
+        convertedPoint = IntPoint([view convertPoint:point fromView:nil]);
+    return convertedPoint;
```
I know you're just moving code around, but how about:
```suggestion
    if (RetainPtr documentView = [view documentView])
        return [documentView convertPoint:point fromView:nil];
    else
        return [view convertPoint:point fromView:nil];
```

aprotyas commented on Source/WebCore/platform/PlatformMouseEvent.h:
```diff
@@ -95,10 +96,10 @@ class PlatformMouseEvent : public PlatformEvent {
-    IntPoint m_movementDelta;
-    IntPoint m_unadjustedMovementDelta;
+    DoublePoint m_position { 0, 0 };
+    DoublePoint m_globalPosition { 0, 0 };
```
These will default initialize because of https://github.com/WebKit/WebKit/pull/48412/files#diff-babff0ffd37a66734c3a64eb13a1c72e02d7df80b05de52689a38f533a300609R114-R115

aprotyas replied:
so feel free to drop the initializers

aprotyas commented on Source/WebKit/Shared/Extensions/WebExtensionTabParameters.h:
```diff
@@ -27,7 +27,7 @@
 #if ENABLE(WK_WEB_EXTENSIONS)
 
-#include "DoubleGeometry.h"
+#include <WebCore/DoubleSize.h>
```
I think this should go after the `WebExtension` headers?

aprotyas commented on Source/WebKit/Shared/Extensions/WebExtensionWindowParameters.h:
```diff
@@ -27,7 +27,7 @@
 #if ENABLE(WK_WEB_EXTENSIONS)
 
-#include "DoubleGeometry.h"
+#include <WebCore/DoubleRect.h>
```
Same comment about the include ordering

aprotyas commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -497,6 +497,21 @@ struct WebCore::CharacterRange {
+    double height()
+}
+
+[WebKitPlatform, CustomHeader] class WebCore::DoubleRect {
```
These aren't actually `CustomHeader` instances though, right?

aprotyas replied:
i.e. they're all in namesake headers?

aprotyas replied:
Also is the `WebKitPlatform` necessary?

lilyspiniolas replied:
(Discussed this elsewhere, `WebKitPlatform` is necessary since `DoubleRect` is needed for `CoreIPCNSValue` serialization, but `CustomHeader` can indeed be removed)

aprotyas commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -135,6 +135,32 @@ void SetUp() final
     }
 };
 
+static NSString *fractionalCoordinatesTestPage = @(R"(
```
constexpr?

aprotyas commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -457,6 +549,8 @@ static void simulateEditContextMenuAppearance(TestWKWebView *webView, CGPoint lo
 }
 
+
+
```
Nit: we can drop these new lines?

aprotyas reviewed (commented):
This looks great! I have a few questions, but for the most part LGTM.

aprotyas commented on Source/WebCore/dom/PointerEvent.cpp:
```diff
@@ -293,4 +296,39 @@ void PointerEvent::receivedTarget()
         predictedEvent->setTarget(this->target());
 }
 
+bool PointerEvent::fractionalCoordinatesAllowedForType(const AtomString& type)
```
This doesn't have to be a class method

aprotyas replied:
i.e.

```cpp
static bool fractionalCoordinatesAllowedForType(const AtomString& type)
{
    ....
}
```

lilyspiniolas replied:
Wouldn't it need to be duplicated in PointerEventIOS then?

aprotyas replied:
oh, my bad, I missed its use there.

aprotyas replied:
Although, given this does not depend on a `PointerEvent` instance state, maybe make it a static class method (or even a free function in the `WebCore` namespace, declared/defined in `PointerEvent.[h|cpp]`). Totally nit though

lilyspiniolas committed:
Enable fractional coordinates for Pointer Events and Touch Events
https://bugs.webkit.org/show_bug.cgi?id=133180
rdar://9564176

Reviewed by Abrar Rahman Protyasha.

Pointer Events and Touch Events now expose fractional coordinate values for
clientX/clientY, pageX/pageY, offsetX/offsetY, and screenX/screenY.
Mouse Events still report whole values. These changes affect all platforms.

For pointer events, events with type `click`, `auxclick`, or `contextmenu`
report floored values.

Create new versions of `DoublePoint`, `DoubleRect`, and `DoubleSize` in WebCore.
These versions retain all existing functionality, and include new additions which
allow the types to be used for pointer event location calculations.

Test coverage:
* Fractional coordinates for mouse and pen pointer events from within a subframe are tested by
  `imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates.html`
  which now passes on macOS, GTK, WPE, and Windows. This test also ensures that
  click events are not fractional.

* Fractional coordinates for touch events from within a subframe are tested by
  `fast/events/touch/touch-fractional-coordinates.html`.

* Fractional coordinates for mouse pointer events on iOS from within a subframe are tested by
  new API tests `FractionalCoordinatesInSimpleIFrame`, `FractionalCoordinatesInRotatedIFrame`
  & `FractionalCoordinatesInScaledIFrame`.

* `imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates.html` now
  checks both client and page coordinates to increase the robustness of the test.

These changes build upon the following proposed patch by Sun Shin (@xingri):
https://github.com/WebKit/WebKit/pull/31867

(Additionally, updated `setContentsToPlatformLayer` to take a `PlatformLayerContainer` instead of a
`PlatformLayer *` due to a safer-cpp issue in order to improve safety.)

* LayoutTests/fast/events/constructors/mouse-event-constructor-expected.txt:
* LayoutTests/fast/events/constructors/mouse-event-constructor.html:
* LayoutTests/fast/events/constructors/wheel-event-constructor-expected.txt:
* LayoutTests/fast/events/constructors/wheel-event-constructor.html:
* LayoutTests/fast/events/touch/ios/touches-client-coords-after-zoom.html:
* LayoutTests/fast/events/touch/resources/frame-touchevent-forwarder.html: Added.
* LayoutTests/fast/events/touch/touch-coords-in-zoom-and-scroll-expected.txt:
* LayoutTests/fast/events/touch/touch-coords-in-zoom-and-scroll.html:
* LayoutTests/fast/events/touch/touch-fractional-coordinates-expected.txt: Added.
* LayoutTests/fast/events/touch/touch-fractional-coordinates.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates.html:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates_mouse-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates_pen-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_fractional_coordinates_untrusted-expected.txt:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/ios/fast/events/touch/ios/touches-client-coords-after-zoom-expected.txt:
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Document+Touch.idl:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::showPlaybackTargetPicker):
* Source/WebCore/dom/DocumentTouch.cpp:
(WebCore::DocumentTouch::createTouch):
* Source/WebCore/dom/DocumentTouch.h:
* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::create):
(WebCore::MouseEvent::MouseEvent):
(WebCore::MouseEvent::initMouseEvent):
* Source/WebCore/dom/MouseEvent.h:
* Source/WebCore/dom/MouseEvent.idl:
* Source/WebCore/dom/MouseEventInit.h:
* Source/WebCore/dom/MouseEventInit.idl:
* Source/WebCore/dom/MouseRelatedEvent.cpp:
(WebCore::MouseRelatedEvent::MouseRelatedEvent):
(WebCore::MouseRelatedEvent::init):
(WebCore::MouseRelatedEvent::initCoordinates):
(WebCore::MouseRelatedEvent::pagePointToClientPoint):
(WebCore::MouseRelatedEvent::pagePointToAbsolutePoint):
(WebCore::MouseRelatedEvent::computeRelativePosition):
(WebCore::MouseRelatedEvent::locationInRootViewCoordinates const):
(WebCore::MouseRelatedEvent::offsetX):
(WebCore::MouseRelatedEvent::offsetY):
(WebCore::MouseRelatedEvent::pageX const):
(WebCore::MouseRelatedEvent::pageY const):
(WebCore::finiteValue):
(WebCore::MouseRelatedEvent::screenX const):
(WebCore::MouseRelatedEvent::screenY const):
(WebCore::MouseRelatedEvent::clientX const):
(WebCore::MouseRelatedEvent::clientY const):
* Source/WebCore/dom/MouseRelatedEvent.h:
(WebCore::MouseRelatedEvent::screenLocation const):
(WebCore::MouseRelatedEvent::windowLocation const):
(WebCore::MouseRelatedEvent::clientLocation const):
(WebCore::MouseRelatedEvent::absoluteLocation const):
(WebCore::MouseRelatedEvent::hasCachedRelativePosition const):
(WebCore::MouseRelatedEvent::offsetLocation const):
(WebCore::MouseRelatedEvent::pageLocation const):
(WebCore::MouseRelatedEvent::setScreenLocation):
(WebCore::MouseRelatedEvent::screenX const): Deleted.
(WebCore::MouseRelatedEvent::screenY const): Deleted.
(WebCore::MouseRelatedEvent::clientX const): Deleted.
(WebCore::MouseRelatedEvent::clientY const): Deleted.
* Source/WebCore/dom/PointerEvent.cpp:
(WebCore::PointerEvent::PointerEvent):
(WebCore::m_fractionalCoordinatesAllowed):
(WebCore::PointerEvent::fractionalCoordinatesAllowedForType):
(WebCore::PointerEvent::adjustedCoordinateForType const):
(WebCore::PointerEvent::offsetX):
(WebCore::PointerEvent::offsetY):
(WebCore::m_isPrimary): Deleted.
* Source/WebCore/dom/PointerEvent.h:
(WebCore::PointerEvent::create):
(WebCore::PointerEvent::fractionalCoordinatesAllowed const):
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/dom/SimulatedClick.cpp:
* Source/WebCore/dom/Touch.cpp:
(WebCore::contentsOffset):
(WebCore::scaledLocation):
(WebCore::Touch::Touch):
(WebCore::Touch::cloneWithNewTarget const):
(WebCore::contentsX): Deleted.
(WebCore::contentsY): Deleted.
* Source/WebCore/dom/Touch.h:
(WebCore::Touch::create):
(WebCore::Touch::clientX const):
(WebCore::Touch::clientY const):
(WebCore::Touch::screenX const):
(WebCore::Touch::screenY const):
(WebCore::Touch::pageX const):
(WebCore::Touch::pageY const):
(WebCore::Touch::webkitRadiusX const):
(WebCore::Touch::webkitRadiusY const):
(WebCore::Touch::absoluteLocation const):
* Source/WebCore/dom/Touch.idl:
* Source/WebCore/dom/TouchEvent.cpp:
(WebCore::TouchEvent::TouchEvent):
* Source/WebCore/dom/TouchEvent.h:
* Source/WebCore/dom/UIEvent.cpp:
(WebCore::UIEvent::screenX const):
(WebCore::UIEvent::screenY const):
(WebCore::UIEvent::pageX const):
(WebCore::UIEvent::pageY const):
(WebCore::UIEvent::clientX const):
(WebCore::UIEvent::clientY const):
* Source/WebCore/dom/UIEvent.h:
* Source/WebCore/dom/UIEvent.idl:
* Source/WebCore/dom/ios/PointerEventIOS.cpp:
(WebCore::PointerEvent::create):
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/dom/wpe/PointerEventWPE.cpp:
(WebCore::PointerEvent::create):
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/html/CheckboxInputType.cpp:
(WebCore::CheckboxInputType::handleMouseDownEvent):
(WebCore::CheckboxInputType::handleMouseMoveEvent):
(WebCore::CheckboxInputType::handleTouchEvent):
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::handleMouseDownEvent):
(WebCore::RangeInputType::handleTouchEvent):
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::defaultEventHandler):
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::InspectorFrontendHost::dispatchEventAsContextMenuEvent):
* Source/WebCore/page/AutoscrollController.cpp:
(WebCore::AutoscrollController::autoscrollTimerFired):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::maybeCreateContextMenu):
* Source/WebCore/page/DebugPageOverlays.cpp:
(WebCore::InteractionRegionOverlay::mouseEvent):
* Source/WebCore/page/DragController.cpp:
(WebCore::DragController::startDrag):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::SyntheticTouchPoint::SyntheticTouchPoint):
(WebCore::EventHandler::handleMousePressEventSingleClick):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::eventMayStartDrag const):
(WebCore::EventHandler::updateSelectionForMouseDrag):
(WebCore::EventHandler::startPanScrolling):
(WebCore::EventHandler::lastKnownMousePosition const):
(WebCore::EventHandler::updateCursor):
(WebCore::documentPointForWindowPoint):
(WebCore::scrollbarForMouseEvent):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseReleaseEvent):
(WebCore::EventHandler::dispatchDragEvent):
(WebCore::EventHandler::updateDragAndDrop):
(WebCore::EventHandler::prepareMouseEvent):
(WebCore::EventHandler::dispatchMouseEvent):
(WebCore::EventHandler::targetPositionInWindowForSelectionAutoscroll const):
(WebCore::EventHandler::handleWheelEventInternal):
(WebCore::EventHandler::sendContextMenuEvent):
(WebCore::EventHandler::dispatchFakeMouseMoveEventSoonInQuad):
(WebCore::EventHandler::hoverTimerFired):
(WebCore::EventHandler::handleDrag):
(WebCore::EventHandler::handleTouchEvent):
(WebCore::EventHandler::setLastKnownMousePosition):
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::convertFromContainingViewToRenderer const):
(WebCore::FrameView::convertFromContainingView const):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::lastKnownMousePositionInView const):
(WebCore::LocalFrameView::absoluteToDocumentPoint const):
(WebCore::LocalFrameView::documentToClientPoint const):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/PageOverlay.cpp:
(WebCore::PageOverlay::mouseEvent):
* Source/WebCore/page/PointerCaptureController.cpp:
(WebCore::PointerCaptureController::dispatchOverOrOutEvent):
(WebCore::PointerCaptureController::dispatchEnterOrLeaveEvent):
(WebCore::PointerCaptureController::dispatchEventForTouchAtIndex):
* Source/WebCore/page/PointerCaptureController.h:
* Source/WebCore/page/RemoteFrameGeometryTransformer.cpp:
(WebCore::RemoteFrameGeometryTransformer::transformToRemoteFrameCoordinates const):
* Source/WebCore/page/RemoteFrameGeometryTransformer.h:
* Source/WebCore/page/ResourceUsageOverlay.cpp:
(WebCore::ResourceUsageOverlay::mouseEvent):
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::tryToBeginDragAtPoint):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::targetPositionInWindowForSelectionAutoscroll const):
* Source/WebCore/page/mac/ImageOverlayControllerMac.mm:
(WebCore::ImageOverlayController::platformHandleMouseEvent):
* Source/WebCore/page/mac/ServicesOverlayController.mm:
(WebCore::ServicesOverlayController::mouseEvent):
* Source/WebCore/platform/PlatformMouseEvent.h:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
(WebCore::PlatformMouseEvent::position const):
(WebCore::PlatformMouseEvent::globalPosition const):
(WebCore::PlatformMouseEvent::movementDelta const):
(WebCore::PlatformMouseEvent::unadjustedMovementDelta const):
* Source/WebCore/platform/PlatformTouchPoint.h:
(WebCore::PlatformTouchPoint::PlatformTouchPoint):
(WebCore::PlatformTouchPoint::screenPos const):
(WebCore::PlatformTouchPoint::pos const):
(WebCore::PlatformTouchPoint::radius const):
(WebCore::PlatformTouchPoint::radiusX const): Deleted.
(WebCore::PlatformTouchPoint::radiusY const): Deleted.
* Source/WebCore/platform/ScrollView.cpp:
(WebCore::ScrollView::viewToContents const):
(WebCore::ScrollView::contentsToView const):
(WebCore::ScrollView::rootViewToContents const):
(WebCore::ScrollView::contentsToRootView const):
(WebCore::ScrollView::windowToContents const):
(WebCore::ScrollView::convertChildToSelf const):
(WebCore::ScrollView::convertSelfToChild const):
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Scrollbar.cpp:
(WebCore::Scrollbar::mouseMoved):
(WebCore::Scrollbar::mouseUp):
(WebCore::Scrollbar::mouseDown):
* Source/WebCore/platform/Widget.cpp:
(WebCore::Widget::convertToRootView const):
(WebCore::Widget::convertFromRootView const):
(WebCore::Widget::convertFromContainingWindow const):
(WebCore::Widget::convertToContainingView const):
(WebCore::Widget::convertFromContainingView const):
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/graphics/DoublePoint.cpp: Copied from Source/WebCore/dom/DocumentTouch.h.
(WebCore::DoublePoint::DoublePoint):
(WebCore::DoublePoint::operator CGPoint const):
(WebCore::DoublePoint::toJSONObject const):
(WebCore::operator<<):
* Source/WebCore/platform/graphics/DoublePoint.h: Added.
(WebCore::DoublePoint::DoublePoint):
(WebCore::DoublePoint::zero):
(WebCore::DoublePoint::isZero const):
(WebCore::DoublePoint::x const):
(WebCore::DoublePoint::y const):
(WebCore::DoublePoint::move):
(WebCore::DoublePoint::moveBy):
(WebCore::DoublePoint::scale):
(WebCore::DoublePoint::scaled const):
(WebCore::DoublePoint::toCG const):
(WebCore::operator+):
(WebCore::operator-):
(WebCore::flooredIntPoint):
(WebCore::roundedIntPoint):
(WebCore::toDoubleSize):
(WTF::LogArgument<WebCore::DoublePoint>::toString):
* Source/WebCore/platform/graphics/DoubleRect.cpp: Copied from Source/WebCore/dom/MouseEventInit.h.
(WebCore::DoubleRect::DoubleRect):
(WebCore::DoubleRect::operator CGRect const):
* Source/WebCore/platform/graphics/DoubleRect.h: Copied from Source/WebCore/dom/DocumentTouch.h.
(WebCore::DoubleRect::DoubleRect):
(WebCore::DoubleRect::location const):
(WebCore::DoubleRect::size const):
(WebCore::DoubleRect::toCG const):
* Source/WebCore/platform/graphics/DoubleSize.cpp: Copied from Source/WebCore/dom/MouseEventInit.h.
(WebCore::DoubleSize::DoubleSize):
(WebCore::DoubleSize::operator CGSize const):
* Source/WebCore/platform/graphics/DoubleSize.h: Copied from Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.h.
(WebCore::DoubleSize::DoubleSize):
(WebCore::DoubleSize::width const):
(WebCore::DoubleSize::height const):
(WebCore::DoubleSize::toCG const):
(WebCore::DoubleSize::scaledBy const):
* Source/WebCore/platform/graphics/FloatPoint.h:
(WebCore::FloatPoint::FloatPoint):
(WebCore::FloatPoint::scaledBy const):
(WebCore::FloatPoint::operator DoublePoint const):
(WebCore::roundedIntPoint):
* Source/WebCore/platform/graphics/FloatSize.h:
(WebCore::FloatSize::scaledBy const):
(WebCore::FloatSize::operator DoubleSize const):
* Source/WebCore/platform/graphics/Font.cpp:
(WebCore::operator<<):
* Source/WebCore/platform/graphics/LayoutPoint.h:
(WebCore::LayoutPoint::LayoutPoint):
(WebCore::LayoutPoint::operator DoublePoint const):
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::GraphicsLayerCA::setContentsToVideoElement):
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
* Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
(WebCore::Font::applyTransforms const):
* Source/WebCore/platform/ios/PlatformEventFactoryIOS.mm:
(WebCore::pointForEvent):
(WebCore::globalPointForEvent):
(WebCore::PlatformWheelEventBuilder::PlatformWheelEventBuilder):
* Source/WebCore/platform/ios/ScrollAnimatorIOS.mm:
(WebCore::ScrollAnimatorIOS::handleTouchEvent):
* Source/WebCore/platform/ios/WidgetIOS.mm:
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.h:
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::pointForEvent):
(WebCore::unadjustedMovementForEvent):
(WebCore::PlatformMouseEventBuilder::PlatformMouseEventBuilder):
(WebCore::PlatformWheelEventBuilder::PlatformWheelEventBuilder):
* Source/WebCore/platform/mac/WidgetMac.mm:
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::resize):
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::panScrollFromPoint):
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::panScroll):
(WebCore::RenderListBox::autoscroll):
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::absoluteToLocal const):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::absoluteToLocal):
* Source/WebCore/testing/MockPageOverlayClient.cpp:
(WebCore::MockPageOverlayClient::mouseEvent):
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.h:
(WebKit::CoreIPCNSShadow::CoreIPCNSShadow):
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.mm:
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.mm:
(WebKit::CoreIPCNSValue::toID const):
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionTab.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionTabParameters.h:
* Source/WebKit/Shared/Extensions/WebExtensionWindow.serialization.in:
* Source/WebKit/Shared/Extensions/WebExtensionWindowParameters.h:
* Source/WebKit/Shared/NativeWebMouseEvent.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
(WebKit::WebKit2PlatformTouchPoint::WebKit2PlatformTouchPoint):
* Source/WebKit/Shared/WebMouseEvent.cpp:
(WebKit::WebMouseEvent::WebMouseEvent):
* Source/WebKit/Shared/WebMouseEvent.h:
(WebKit::WebMouseEvent::WebMouseEvent):
(WebKit::WebMouseEvent::position const):
(WebKit::WebMouseEvent::setPosition):
(WebKit::WebMouseEvent::globalPosition const):
(WebKit::WebMouseEvent::unadjustedMovementDelta const):
* Source/WebKit/Shared/WebPlatformTouchPoint.cpp:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
* Source/WebKit/Shared/WebTouchEvent.h:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
(WebKit::WebPlatformTouchPoint::locationInRootView const):
(WebKit::WebPlatformTouchPoint::locationInViewport const):
(WebKit::WebTouchEvent::WebTouchEvent):
(WebKit::WebTouchEvent::position const):
(WebKit::WebPlatformTouchPoint::screenPosition const):
(WebKit::WebPlatformTouchPoint::position const):
(WebKit::WebPlatformTouchPoint::radius const):
* Source/WebKit/Shared/graphics/DoubleGeometry.h: Removed.
* Source/WebKit/Shared/graphics/DoubleGeometry.serialization.in: Removed.
* Source/WebKit/Shared/gtk/NativeWebMouseEventGtk.cpp:
(WebKit::NativeWebMouseEvent::NativeWebMouseEvent):
* Source/WebKit/Shared/gtk/WebEventFactory.cpp:
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/Shared/gtk/WebEventFactory.h:
* Source/WebKit/Shared/ios/NativeWebMouseEventIOS.mm:
(WebKit::NativeWebMouseEvent::NativeWebMouseEvent):
* Source/WebKit/Shared/ios/NativeWebTouchEventIOS.mm:
(WebKit::positionForCGPoint):
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(webkitWebViewBaseButtonPressed):
(webkitWebViewBaseButtonReleased):
(webkitWebViewBaseEnter):
(webkitWebViewBaseMotion):
(webkitWebViewBaseLeave):
(webkitWebViewBaseSynthesizeMouseEvent):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateTouchEventTracking):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _handleTouchActionsForTouchEvent:]):
* Source/WebKit/UIProcess/ios/WKMouseInteraction.mm:
(-[WKMouseInteraction createMouseEventWithType:wasCancelled:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPITabsCocoa.mm:
(WebKit::toWebAPI):
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePageOverlay.cpp:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::handleMouseEvent):
(WebKit::PDFPlugin::handleContextMenuEvent):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::lastKnownMousePositionInView const):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFDataDetectorOverlayController.mm:
(WebKit::PDFDataDetectorOverlayController::handleMouseEvent):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::handleMouseEvent):
(WebKit::UnifiedPDFPlugin::handleContextMenuEvent):
(WebKit::UnifiedPDFPlugin::createContextMenu const):
(WebKit::UnifiedPDFPlugin::handleSyntheticClick):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::handleContextMenuEvent):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::rectForElementAtInteractionLocation const):
(WebKit::WebPage::focusedElementInformation):
* Source/WebKit/WebProcess/WebPage/mac/PageBannerMac.mm:
(WebKit::PageBanner::mouseEvent):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::shouldDelayWindowOrderingEvent):
(WebKit::WebPage::requestAcceptsFirstMouse):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(TEST_F(iOSMouseSupport, FractionalCoordinatesInSimpleIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInRotatedIFrame)):
(TEST_F(iOSMouseSupport, FractionalCoordinatesInScaledIFrame)):

Canonical link: https://commits.webkit.org/298383@main
